core/trino-spi/src/main/java/io/trino/spi/connector/JoinCondition.java

```diff
@@ -13,36 +13,113 @@
  */
 package io.trino.spi.connector;
 
+import io.trino.spi.expression.Call;
 import io.trino.spi.expression.ConnectorExpression;
+import io.trino.spi.expression.FunctionName;
+import io.trino.spi.expression.StandardFunctions;
+import io.trino.spi.expression.Variable;
 
+import java.util.ArrayDeque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
+import java.util.stream.Stream;
 
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toUnmodifiableMap;
 
+@Deprecated
 public final class JoinCondition
 {
+    @Deprecated
     public enum Operator
     {
-        EQUAL("="),
-        NOT_EQUAL("<>"),
-        LESS_THAN("<"),
-        LESS_THAN_OR_EQUAL("<="),
-        GREATER_THAN(">"),
-        GREATER_THAN_OR_EQUAL(">="),
-        IS_DISTINCT_FROM("IS DISTINCT FROM");
+        EQUAL("=", StandardFunctions.EQUAL_OPERATOR_FUNCTION_NAME),
+        NOT_EQUAL("<>", StandardFunctions.NOT_EQUAL_OPERATOR_FUNCTION_NAME),
+        LESS_THAN("<", StandardFunctions.LESS_THAN_OPERATOR_FUNCTION_NAME),
+        LESS_THAN_OR_EQUAL("<=", StandardFunctions.LESS_THAN_OR_EQUAL_OPERATOR_FUNCTION_NAME),
+        GREATER_THAN(">", StandardFunctions.GREATER_THAN_OPERATOR_FUNCTION_NAME),
+        GREATER_THAN_OR_EQUAL(">=", StandardFunctions.GREATER_THAN_OR_EQUAL_OPERATOR_FUNCTION_NAME),
+        IS_DISTINCT_FROM("IS DISTINCT FROM", StandardFunctions.IS_DISTINCT_FROM_OPERATOR_FUNCTION_NAME),
+        /**/;
+
+        private static final Map<FunctionName, Operator> byFunctionName = Stream.of(values())
+                .collect(toUnmodifiableMap(operator -> operator.callFunctionName, identity()));
 
         private final String value;
+        private final FunctionName callFunctionName;
 
-        Operator(String value)
+        Operator(String value, FunctionName callFunctionName)
         {
             this.value = value;
+            this.callFunctionName = callFunctionName;
         }
 
         public String getValue()
         {
             return value;
         }
+
+        public Operator flip()
+        {
+            return switch (this) {
+                case EQUAL, NOT_EQUAL, IS_DISTINCT_FROM -> this;
+                case LESS_THAN -> GREATER_THAN;
+                case LESS_THAN_OR_EQUAL -> GREATER_THAN_OR_EQUAL;
+                case GREATER_THAN -> LESS_THAN;
+                case GREATER_THAN_OR_EQUAL -> LESS_THAN_OR_EQUAL;
```

Comment:
Add throw on `default`?

Reply:
All options should be covered, so `default` is redundant.
Lack of `default` makes it a compilation problem if i forget to cover an option (or if a new option is added without this method being updated).

Reply:
Oh ok, didn't know it won't compile

Reply:
[switch expressions ](https://openjdk.org/jeps/361):

```
Exhaustiveness
The cases of a switch expression must be exhaustive; for all possible values there must be a matching switch label. (Obviously switch statements are not required to be exhaustive.)
```

Reply:
This is one of the JDK changes that I see really useful - no need for static code analysis to find missing switch cases :)

Reply:
Yep. And since JLS rules are now consistent with the desired state (no fake handling of "default" where it's not intentional), it's so much more concise. With ordinary switch statements, you can use static code analysis (like error-prone), but you still need _some_ handling of the "default" case, as required by the JLS.